### DISCUSSION TITLE
Investigate Postgres support

### DISCUSSION THREAD

**Discussion by aaronbushnell** [trendyminds/isolate#35] [Ideas]
Hey @aaronbushnell, isolate is really good plugin... but I have problem with pgslq database, I assume you made sql queries for msyql, do you have in mind to build it for pgsql?

_Originally posted by @Kreshimir in https://github.com/trendyminds/isolate/issues/3#issuecomment-491711278_

**Comment by aaronbushnell** [comment]
@Kreshimir Are you having issues in Postgres? I believe all DB functionality is using the Craft API so it should be compatible with whatever Craft is.

**Comment by Kreshimir** [comment]
@aaronbushnell Yes, only with postgres 
Error -> 
SQLSTATE[42703]: Undefined column: 7 ERROR: column con.elementid does not exist
LINE 3: LEFT JOIN "content" "con" ON con.elementId=ent.id
^
HINT: Perhaps you meant to reference the column "con.elementId".
The SQL being executed was: SELECT "ent"."id", "con"."title", "sec"."handle", "con"."siteId"
FROM "entries" "ent"
LEFT JOIN "content" "con" ON con.elementId=ent.id
LEFT JOIN "sections" "sec" ON sec.id=ent.sectionId
WHERE "sec"."id"=7
ORDER BY "con"."title"

With mysql works perfect.
Any idea?

**Comment by aaronbushnell** [comment]
Very odd! I'll investigate as soon as possible

**Comment by Kreshimir** [comment]
@aaronbushnell thank you!

**Comment by etienneasselin** [comment]
Queries in postgres must be like 
```        $entries = $query->select(['"ent"."id"', '"con".title', '"sec"."handle"', '"con"."siteId"'])
            ->from("{{%entries}} ent")
            ->leftJoin("{{%content}} con", '"con"."elementId"="ent"."id"')
            ->leftJoin("{{%sections}} sec", '"sec"."id"="ent"."sectionId"')
            ->filterWhere(['"sec"."id"' => $sectionId])
            ->orderBy('"con"."title"')
            ->all();`

**Comment by Kreshimir** [comment]
@etienneasselin works perfect, thanks :)

**Comment by Kreshimir** [comment]
@etienneasselin now I have problem with this query ->
$entries = $query
            ->select([
                "ent.id",
                "con.title",
                "sec.handle",
                "struc.level",
                "con.siteId",
            ])
            ->from("{{%structureelements}} struc")
            ->leftJoin("{{%elements}} elems", "struc.elementId = elems.id")
            ->leftJoin("{{%content}} con", "con.elementId=struc.elementId")
            ->leftJoin("{{%entries}} ent", "con.elementId=ent.id")
            ->leftJoin("{{%sections}} sec", "sec.id=ent.sectionId")
            ->where(["struc.structureId" => $structure->id])
            ->andWhere("con.title IS NOT NULL")
            ->orderBy([
                "lft" => SORT_ASC,
            ])
            ->all();
any idea?

**Comment by aaronbushnell** [comment]
Hello! We've moved all open Issues to **Discussions**.

Unfortunately, due to client workload we often find ourselves unable to implement new features or resolve issues in a timely manner. Switching to Discussions allows better categorization to separate "Feature Requests" from bugs. It also allows the community to better suggest/provide solutions when we're unable to do so.

Hope Isolate continues to be useful for you!
